File: analytics/src/main/java/com/bluetriangle/analytics/NoOpLogger.java
package com.bluetriangle.analytics;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * A logger that does not actually log anything
 */
final public class NoOpLogger extends Logger {

    private static final NoOpLogger instance = new NoOpLogger();

    public static NoOpLogger getInstance() {
        return instance;
    }

    private NoOpLogger() {}

    @Override
    void log(int logLevel, @NonNull String message, @Nullable Object... args) {}

    @Override
    void log(int logLevel, @Nullable Throwable throwable, @NonNull String message, @Nullable Object... args) {}
}

File: analytics/src/main/java/com/bluetriangle/analytics/NoOpLogger.kt
package com.bluetriangle.analytics

import com.bluetriangle.analytics.NoOpLogger

/**
 * A logger that does not actually log anything
 */
internal class NoOpLogger private constructor() : Logger() {
    override fun log(logLevel: Int, message: String) {}
    override fun log(logLevel: Int, throwable: Throwable?, message: String) {}

    companion object {
        val instance = NoOpLogger()
    }
}